File: src/main/java/net/andrecarbajal/naviMusic/commands/music/PauseCommand.java
package net.andrecarbajal.naviMusic.commands.music;

import net.andrecarbajal.naviMusic.audio.MusicService;
import net.andrecarbajal.naviMusic.commands.SlashCommand;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import org.springframework.stereotype.Component;

@Component
public class PauseCommand extends SlashCommand {
    private final MusicService musicService;

    public PauseCommand(MusicService musicService) {
        super("pause", "Pauses music", Category.MUSIC);
        this.musicService = musicService;
    }

    @Override
    public void onCommand(SlashCommandInteractionEvent event) {
        if (noVoiceChannelCheck(event)) return;
        event.deferReply().queue();

        musicService.pause(event.getChannel().asTextChannel()).editReply(event);
    }
}

File: src/main/kotlin/net/andrecarbajal/naviMusic/commands/music/PauseCommand.kt
package net.andrecarbajal.naviMusic.commands.music

import net.andrecarbajal.naviMusic.audio.MusicService
import net.andrecarbajal.naviMusic.commands.SlashCommand
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import org.springframework.stereotype.Component

@Component
class PauseCommand(private val musicService: MusicService) : SlashCommand("pause", "Pauses music", Category.MUSIC) {

    override fun onCommand(event: SlashCommandInteractionEvent) {
        if (noVoiceChannelCheck(event)) return
        event.deferReply().queue()

        val textChannel = event.channel.asTextChannel()
        musicService.pause(textChannel).editReply(event)
    }
}
